eslint-plugin-prettier.js

```diff
@@ -0,0 +1,306 @@
+  // fast-diff returns the differences between two texts as a series of
+  // INSERT, DELETE or EQUAL operations. The results occur only in these
+  // sequences:
+  //           /-> INSERT -> EQUAL
+  //    EQUAL |           /-> EQUAL
+  //           \-> DELETE |
+  //                      \-> INSERT -> EQUAL
+  // Instead of reporting issues at each INSERT or DELETE, certain sequences
+  // are batched together and are reported as a friendlier "replace" operation:
+  // - A DELETE immediately followed by an INSERT.
+  // - Any number of INSERTs and DELETEs where the joining EQUAL of one's end
+  // and another's beginning does not have line endings (i.e. issues that occur
+  // on contiguous lines).
+
+  const source = context.getSource();
+  const results = diff(source, prettierSource);
+
+  const batch = [];
+  let offset = 0; // NOTE: INSERT never advances the offset.
+  while (results.length) {
+    const result = results.shift();
+    const op = result[0];
+    const text = result[1];
+    switch (op) {
+      case diff.INSERT:
+      case diff.DELETE:
+        batch.push(result);
+        break;
+      case diff.EQUAL:
+        if (results.length) {
+          if (batch.length) {
+            if (LINE_ENDING_RE.test(text)) {
+              flush();
+              offset += text.length;
+            } else {
+              batch.push(result);
+            }
+          } else {
+            offset += text.length;
+          }
+        }
+        break;
+      default:
+        throw new Error(`Unexpected fast-diff operation "${op}"`);
+    }
+    if (batch.length && !results.length) {
+      flush();
+    }
+  }
+
+  function flush() {
+    let aheadDeleteText = '';
+    let aheadInsertText = '';
+    while (batch.length) {
+      const next = batch.shift();
+      const op = next[0];
+      const text = next[1];
+      switch (op) {
+        case diff.INSERT:
+          aheadInsertText += text;
+          break;
+        case diff.DELETE:
+          aheadDeleteText += text;
+          break;
+        case diff.EQUAL:
+          aheadDeleteText += text;
+          aheadInsertText += text;
+          break;
+      }
+    }
+    if (aheadDeleteText && aheadInsertText) {
+      reportReplace(context, offset, aheadDeleteText, aheadInsertText);
+    } else if (!aheadDeleteText && aheadInsertText) {
+      reportInsert(context, offset, aheadInsertText);
+    } else if (aheadDeleteText && !aheadInsertText) {
+      reportDelete(context, offset, aheadDeleteText);
+    }
+    offset += aheadDeleteText.length;
+  }
+}
+
+/**
+ * Reports an "Insert ..." issue where text must be inserted.
+ * @param {RuleContext} context - The ESLint rule context.
+ * @param {number} offset - The source offset where to insert text.
+ * @param {string} text - The text to be inserted.
+ * @returns {void}
+ */
+function reportInsert(context, offset, text) {
+  const pos = getLocFromIndex(context, offset);
+  const range = [null, offset];
```

Comment:
I'm not sure `null` is documented as a valid value in a fix range. Can you replace this with `[offset, offset]` or something like that instead?

(Admittedly, the design of `insertTextAfterRange` does make the first value in the range useless, but I don't want this to break if ESLint starts validating ranges later on by asserting that both values are numbers.)